src/algorithms/panocplus.jl

```diff
@@ -178,7 +177,6 @@ function Base.iterate(iter::PANOCplusIteration{R}, state::PANOCplusState) where
             break
         end
         state.tau = tau_backtracks >= iter.max_backtracks - 1 ? R(0) : state.tau / 2
-        tau_backtracks += 1
```

Comment:
This value is used in the lines immediately above this one, in some checks: if the value is instead updated earlier (at the top of the loop) probably the inequalities involving it need to be adjusted?

Reply:
The algorithm is identical in this regard, since it starts always with `can_update_direction = true` and so it takes `tau_backtracks = 0`. Then the update `tau_backtracks += 1`, or the reset, is effectively performed after the checks you mention.